refactor: Refactor/clean AsyncTable hooks and add documentation (#2115)
RedHatInsights/compliance-frontend: src/Frameworks/AsyncTableTools/docs/filter-serialiser.md

To be able and filter in tables in Compliance and translate these filters into a Compliance scoped search filter string,
we make use of the "serialiser" feature of the `AsyncTableToolsTable` for filters.

A default `filterSerialiser` function, that can transform the values most of the filters into a scoped search query, provided by the `useFilterConfig` hook and the [ConditionalFilter](https://github.com/RedHatInsights/frontend-components/blob/master/packages/components/doc/conditionalFilter.md) component.
For filters that can not be transformed out of the box can provide a custom serialisation function.

## Filter by attribute

A basic text filter to search the `name` attribute in the API would look like this:

```
const policyNameFilter = [
  {
    label: 'Policy name',
    type: 'text',
    filterAttribute: 'name',
  }
];
```

> The `lable` property simple defines what the filter is called in the `ConditionalFilter`.

Reviewer: ```suggestion
The `label` property simple defines what the filter is called in the `ConditionalFilter`.
```

> The `type` property defines what kind of filter it is, this is used by `useFilterConfig` itself and the `filterSerialiser`.

Reviewer: We could mention or link it to PrimaryToolbar filter types https://github.com/RedHatInsights/frontend-components/blob/master/packages/components/doc/conditionalFilter.md

> The `filterAttribute` property tells the `filterSerialiser`, and the text filter serialise function, which attribute to construct the scoped search query for.
> The "attributes" and what can be used in a scoped search query can be found in the API documentation of that endpoint in the `filter` parameter description.

Reviewer: nit-pick: a bit difficult to read, so, perhaps
```suggestion
The `filterAttribute` property defines which attribute to construct the scoped search query for. It is used by `filterSerialiser` and the text filter serialise function.

The "attributes" and what can be used in a scoped search query can be found in the API documentation of that endpoint in the `filter` parameter description.
```

Also, would be nice to add just one simple example in the style: "For example, we have a filter with the title "Awesome filter"; with `filterAttribute` defined as "awesome_f", the filter value will be translated into "awesome_f=foobar"... if you know what I mean 😄

## Filter with filterSerialiser

If a filter requires a more complex query, or just needs to query two attributes,
A filter can provide a `filterSerialiser` function, which will then be used instead of the default one.

> An filter with it's own serialiser would look like this:

Reviewer: ```suggestion
A filter with its own custom serialiser would look like this:
```
